DOC: update array API standard version in compatibility page (#28626)
numpy/numpy: doc/source/reference/array_api.rst

********************************

NumPy's main namespace as well as the `numpy.fft` and `numpy.linalg` namespaces
are compatible with the
`2023.12 version <https://data-apis.org/array-api/2023.12/index.html>`__
of the Python array API standard.

NumPy aims to implement support for the

> `2024.12 version <https://data-apis.org/array-api/2024.12/index.html>`__
> and future versions of the standard - assuming that those future versions can be
> upgraded to given NumPy's

Author: ```suggestion
`2024.12 version <https://data-apis.org/array-api/2024.12/index.html>`__
(in progress) and future versions of the standard - assuming that those future
versions can be upgraded to given NumPy's
```

********************************

> NumPy's main namespace as well as the `numpy.fft` and `numpy.linalg` namespaces

Author: ```suggestion
The NumPy 2.3.0 main namespace as well as the `numpy.fft` and `numpy.linalg`
```

> are compatible with the

Author: ```suggestion
namespaces are compatible with the
```

`2024.12 version <https://data-apis.org/array-api/2024.12/index.html>`__

> of the Python array API standard.

Reviewer: Maybe `The NumPy 2.3.0 main ...` so that it is clear which NumPy version has compatibility.

Author: Done!

********************************

NumPy's main namespace as well as the `numpy.fft` and `numpy.linalg` namespaces
are compatible with the

> `2023.12 version <https://data-apis.org/array-api/2023.12/index.html>`__

Author: ```suggestion
`2024.12 version <https://data-apis.org/array-api/2024.12/index.html>`__
```

of the Python array API standard.

> NumPy aims to implement support for the

Author: ```suggestion
NumPy aims to implement support for the future versions of the standard
```

> `2024.12 version <https://data-apis.org/array-api/2024.12/index.html>`__
> (in progress) and future versions of the standard - assuming that those future
> versions can be upgraded to given NumPy's

Author: ```suggestion
- assuming that those future versions can be upgraded to given NumPy's
```

    NumPy 1.22.0 was the first version to include support for the array API
    standard, via a separate ``numpy.array_api`` submodule. This module was
    marked as experimental (it emitted a warning on import) and removed in

>     NumPy 2.0 because full support (2022.12 version [1]_) was included in the main namespace (with
>     2023.12 version support in NumPy 2.1).

Author: ```suggestion
    NumPy 2.0 because full support (2022.12 version [1]_) was included in
    the main namespace (with 2023.12 version support in NumPy 2.1).
```

>     the main namespace (with 2023.12 version support in NumPy 2.1).

Author: ```suggestion
    the main namespace.
```
